### ISSUE TITLE
Update editor to support boxes

### ISSUE THREAD

**Issue by NuclearRedeye** [libero/editor#541]
# Design
See Epic

# DoD
- [x] Can display a box.
- [x] Can insert a box.
- [x] Can edit the contents of a box.
- [x] Can delete a box.

**Comment by fred-atherden** [comment]
Thanks @ilya-zhuravel, box functionality looks good to me. Will leave @NickDuf to comment on margin/padding if there are any issues there.

I notice that now there are some issues with text wrapping in the main text though
![Screenshot 2020-10-09 at 11.59.55.png](https://images.zenhubusercontent.com/5d5a77bfe56ed603cbb191f6/01d7c486-7af7-462b-8bbf-bcfb93b30fea)

(I missed standup today, so sorry in advance if this was already covered/mentioned).

**Comment by fred-atherden** [comment]
Text wrapping issues are in the box as well (not just main text).

**Comment by NickDuf** [comment]
@ilya-zhuravel couple of items for feedback 

- Inserting the box doesn't add it where the users cursor is
- When adding a new box I can't type into it (the cursor moves out of the box)
- I think we agreed to remove the trash button for removal and use the backspace?

**Comment by ilya-zhuravel** [comment]
@NickDuf 
Removing the box with a button turned out to be easier
When you say the box is not inserted where the cursor where exactly is it added and what is the expected behavior?

**Comment by NickDuf** [comment]
@ilya-zhuravel at present if I return a new line after a paragraph so that I can put my box there, when I use the insert control the box appears on a new line after the line I have created. It's almost correct, I just don't want the line above the box that has been added.

Here are some grabs.

Step 1
![Screenshot 2020-10-12 at 10.38.58.png](https://images.zenhubusercontent.com/5d5a52c42298e3000103df57/88e4da93-704f-4411-b5e9-98feb1a14b47) 

Step 2
![Screenshot 2020-10-12 at 10.39.25.png](https://images.zenhubusercontent.com/5d5a52c42298e3000103df57/38c8eff8-33fb-4232-a252-32b04cdeae07)
